doc(blog): release 0.42.0 blog post (#1500)
falcosecurity/falco-website: content/en/blog/falco-0-42.0/index.md

```yaml
capture:
  # -- Set to true to enable event capturing.
  enabled: false
  # -- Prefix for capture files. Falco appends a timestamp and event number to ensure unique filenames.
  path_prefix: /tmp/falco
  # -- Capture mode. Can be "rules" or "all_rules".
  mode: rules
  # -- Default capture duration in milliseconds if not specified in the rule.
  default_duration: 5000
```

**Learn more at KubeCon + CloudNativeCon North America 2025:**
- [Project Lightning Talk: When Falco Spots Trouble, The Shark Swims In](https://kccncna2025.sched.com/event/27d4o/project-lightning-talk-when-falco-spots-trouble-the-shark-swims-in-gerald-combs-falco-core-maintainer) - Gerald Combs, Falco Core Maintainer
- [Beyond the Cloud(s): Falco's Ascent in Performance and Deep Visibility](https://kccncna2025.sched.com/event/27No0/beyond-the-clouds-falcos-ascent-in-performanc[…]eep-visibility-leonardo-grasso-leonardo-di-giovanna-sysdig) - Leonardo Grasso & Leonardo Di Giovanna, Sysdig

### Drop enter initiative

We've just shipped a major performance improvement: syscall enter events have been completely removed from our event pipeline.

> In Falco, each system call traditionally used to generate two events: an enter event when the syscall kernel processing starts (i.e. before its arguments are processed) and an exit event when the kernel processing completes. Since now we collect all relevant information on exit events, we could drop enter events generation and processing.

Reviewer: ```suggestion
In Falco, each system call traditionally used to generate two events: an enter event when syscall kernel processing starts (i.e., before its arguments are processed) and an exit event when the kernel processing completes. Now that we collect all relevant information on exit events, we can drop the generation and processing of enter events.
```

   // Required: no
   // Arguments:
   // - s: the plugin state, returned by init(). Can be NULL.
   // Return value: the event schema version string, in the following format:
   //       "<major>.<minor>.<patch>", e.g. "4.0.0".
   //       If the function is not implemented or NULL is returned, the plugin is assumed to be
   //       compatible with schema version 3.0.0.
   //
   const char* (*get_required_event_schema_version)(ss_plugin_t* s);
} plugin_api;
```

For more details, see:
- [Plugin system event schema versioning proposal](https://github.com/falcosecurity/libs/blob/master/proposals/20250923-plugin-system-event-schema-versioning.md)

### Thread table auto-purging configuration

We’ve added a few new `falco_libs` configurations for advanced users who want finer control over Falco’s performance and resource usage.
It introduces tunable parameters for Falco’s internal thread table, which tracks active threads:

* `thread_table_size` defines the maximum number of entries.
* `thread_table_auto_purging_interval_s` controls how often stale threads are cleaned up.
* `thread_table_auto_purging_thread_timeout_s` sets how long inactive threads are kept before removal.

> These options let you balance memory efficiency, CPU usage, and state accuracy, with related metrics (`n_drops_full_threadtable`, `n_store_evts_drops`) available to guide tuning.

Reviewer: ```suggestion
We've added a few new `falco_libs` configurations for advanced users who want finer control over Falco's performance and resource usage.
It introduces tunable parameters for Falco's internal thread table, which tracks active threads:

* `thread_table_size` defines the maximum number of entries.
* `thread_table_auto_purging_interval_s` controls how often stale threads are cleaned up.
* `thread_table_auto_purging_thread_timeout_s` sets how long inactive threads are kept before removal.

These options let you balance memory efficiency, CPU usage, and state accuracy, with related metrics (`n_drops_full_threadtable`, `n_store_evts_drops`) available to guide tuning.
```
